File: trunk/src/main/java/net/esper/pattern/guard/GuardEnum.java
package net.esper.pattern.guard;

/**
 * Enum for all build-in guards.
 */
public enum GuardEnum
{
    /**
     * Timer guard.
     */
    TIMER_WITHIN("timer", "within", TimerWithinGuardFactory.class);

    private final String namespace;
    private final String name;
    private final Class clazz;

    GuardEnum(String namespace, String name, Class clazz)
    {
        this.namespace = namespace;
        this.name = name;
        this.clazz = clazz;
    }

    /**
     * Returns the namespace name.
     * @return namespace name
     */
    public String getNamespace()
    {
        return namespace;
    }

    /**
     * Returns name.
     * @return short name
     */
    public String getName()
    {
        return name;
    }

    /**
     * Gets the implementation class.
     * @return implementation class
     */
    public Class getClazz()
    {
        return clazz;
    }

    /**
     * Returns the enum for the given namespace and name.
     * @param namespace - guard namespace
     * @param name - guard name
     * @return enum
     */
    public static GuardEnum forName(String namespace, String name)
    {
        for (GuardEnum guardEnum : GuardEnum.values())
        {
            if ((guardEnum.getNamespace().equals(namespace)) && (guardEnum.getName().equals(name)))
            {
                return guardEnum;
            }
        }

        return null;
    }
}

File: NEsper/pattern/guard/GuardEnum.cs
using System;

namespace net.esper.pattern.guard
{
    /// <summary> 
    /// Enum for all build-in guards.
    /// </summary>

    public class GuardEnum
    {
        /// <summary>
        /// Timer guard.
        /// </summary>
        
        public static readonly GuardEnum TIMER_WITHIN = new GuardEnum("timer", "within", typeof(TimerWithinGuardFactory));

        /// <summary>
        /// All values available through this pseudo-enum
        /// </summary>
        
        public static readonly GuardEnum[] Values = new GuardEnum[]
        {
        	TIMER_WITHIN
        };
        
        private readonly String nspace;
        private readonly String name;
        private readonly Type clazz;

        /// <summary>
        /// Initializes a new instance of the <see cref="GuardEnum"/> class.
        /// </summary>
        /// <param name="nspace">The nspace.</param>
        /// <param name="name">The name.</param>
        /// <param name="clazz">The clazz.</param>
        GuardEnum(String nspace, String name, Type clazz)
        {
            this.nspace = nspace;
            this.name = name;
            this.clazz = clazz;
        }

        /// <summary>
        /// Gets the namespace.
        /// </summary>
        /// <value>The namespace.</value>
        
        public String Namespace
        {
            get { return nspace; }
        }

        /// <summary>
        /// Gets the name.
        /// </summary>
        /// <value>The name.</value>

        public String Name
        {
            get { return name; }
        }

        /// <summary>
        /// Gets the implementation clazz.
        /// </summary>
        /// <value>The implementation clazz.</value>
        
        public Type Clazz
        {
            get { return clazz; }
        }

        /// <summary>Returns the enum for the given namespace and name.</summary>
        /// <param name="nspace">guard namespace</param>
        /// <param name="name">guard name</param>
        /// <returns>enum</returns>

        public static GuardEnum forName(String nspace, String name)
        {
            foreach (GuardEnum guardEnum in GuardEnum.Values)
            {
                if ((guardEnum.Namespace.Equals(nspace)) && (guardEnum.Name.Equals(name)))
                {
                    return guardEnum;
                }
            }

            return null;
        }
    }
}
